### PR TITLE
Login QA test sheet

### PR DESCRIPTION
**Pull request by phillipm**

### CODE REVIEW

File: features/login.feature [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+@QA
+Scenario: Ensure login shows proper error messages
+    Then I install the ccz app at "test_list_search.ccz"
+
+    # normal login
+    Then I login with username "user_with_no_data" and password "123"
+    # TODO: consider counting the home screen buttons
+    Then I logout
+
+    # login in portrait mode
```
Comment (amstone326) [inline features/login.feature]: Can you change this comment to say "landscape mode". Also curious why the step definition we have defined doesn't work in landscape?

File: features/login.feature [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    Then I see the text "Invalid Username or Password"
+
+    # login offline
+    Then I toggle airplane mode
+    Then I login with username "user_with_no_data" and password "123"
+    Then I logout
+
+    # try offline login with bad password
+    Then I login with username "user_with_no_data" and password "bad pass", without waiting for completion
+    Then I see the text "Either the password"
```
Comment (amstone326) [inline features/login.feature]: Might be worth including a bit more of the text here, just for context if nothing else

File: features/login.feature [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+    # try offline login with bad username
+    Then I login with username "fake user" and password "bad pass", without waiting for completion
+    Then I see the text "Reach Server"
```
Comment (amstone326) [inline features/login.feature]: same comment as above

**Comment by phillipm** [comment]
@amstone326 I've addressed your comments

File: features/login.feature [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    Then I clear input field number 1
+    Then I close the keyboard
+    Then I clear input field number 2
+    Then I close the keyboard
+    Then I enter "user_with_no_data" into input field number 1
+    Then I close the keyboard
+    Then I enter "123" into input field number 2
+    Then I close the keyboard
+    Then I press login
+    Then I wait for progress
```
Comment (amstone326) [inline features/login.feature]: @phillipm The test is failing for me at this line with ```map android.widget.ProgressBar, query failed because: Could not detect visibility of sh.calaba.instrumentationbackend.query.ui.UIObjectView@5da4cbe
       (RuntimeError)```

File: features/login.feature [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+    # TODO: consider counting the home screen buttons
+    Then I logout
+    Then I rotate to portrait
+    Then I wait
+
+    # login in demo mode
+    Then I select "Enter as Demo User" menu item
```
Comment (amstone326) [inline features/login.feature]: Heads up that in 2.31, this text has changed to "Enter Practice Mode". I'm not sure how we want to handle these sorts of changes going forward.

File: features/login.feature (hunk 4 above)
Comment (phillipm) [inline features/login.feature]: I think I resolved this. If not, might also require updating to calabash 0.8.2

**Comment by phillipm** [comment]
@amstone326 updated

File: features/login.feature [hunk 6]
```diff
@@ -55,10 +55,19 @@ Scenario: Ensure login shows proper error messages
 
     # try offline login with bad password
     Then I login with username "user_with_no_data" and password "bad pass", without waiting for completion
-    Then I see the text "Either the password you entered was incorrect, or CommCare couldn't reach the server"
+    # Check that you see "Either the password you entered was incorrect, or
```
Comment (amstone326) [inline features/login.feature]: @phillipm Why did you change these?

Comment (phillipm) [inline features/login.feature]: I couldn't get the test to pass for the full string. Something about the `'` character or word wrapping.

File: features/login.feature (hunk 4 above)
Comment (amstone326) [inline features/login.feature]: Resolved for me as well now

File: features/login.feature [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    Then I press login
+    Then I wait for progress
+
+    # TODO: consider counting the home screen buttons
+    Then I logout
+    Then I rotate to portrait
+    Then I wait
+
+    # login in demo mode
+    Then I select "Enter as Demo User" menu item
```
Comment (amstone326) [inline features/login.feature]: Hey so this step is now failing for me (I couldn't get to this point before because of the failure above). The reason is that the call to `Then I logout` on line 28 doesn't actually work, seemingly because in landscape mode, the logout button is not visible, and I don't think the way `Then I logout` is implemented provisions for that. Are you not experiencing the same issue when you run it? Maybe your phone shows more in landscape mode than mine does, but I think the way we look for home screen buttons to press needs to be more reobust

File: features/login.feature [hunk 8]
```diff
@@ -0,0 +1,75 @@
+Feature: Login behavior
+@QA
+Scenario: Test login logic: error handling, demo mode, and normal behavior
+    Then I install the ccz app at "test_list_search.ccz"
+
+    # normal login
+    Then I login with username "user_with_no_data" and password "123"
+    # TODO: consider counting the home screen buttons
+    Then I verify that all home buttons are present
```
Comment (amstone326) [inline features/login.feature]: Can you call this after each login? I know it might seem like overkill but there also isn't any reason not to, and it keeps with the original test plan

**Comment by amstone326** [comment]
@phillipm Heads up this also has conflicts with master now

File: features/login.feature (hunk 7 above)
Comment (phillipm) [inline features/login.feature]: Updated it to pan down to the logout button

File: features/login.feature (hunk 8 above)
Comment (phillipm) [inline features/login.feature]: 👍

File: features/login.feature (hunk 7 above)
Comment (amstone326) [inline features/login.feature]: Same issue is happening in landscape mode with the call to `Then I verify that all home buttons are present` :(

**Comment by amstone326** [comment]
Everything is passing now but the issue I just mentioned above!

File: features/login.feature (hunk 7 above)
Comment (phillipm) [inline features/login.feature]: Should be good to go now
